### ISSUE TITLE
RedundantCondition seen after a foreach loop with break and continue

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#377]
Code like this is rare:

Observed: RedundantCondition
Expected: No warnings

```php
<?php
function _load(array $data) : bool {
    $changed = false;
    foreach ($data as $value) {
        if ($value > 0) {
            $changed = true;
            continue;
        }

        break;
    }   
    if ($changed) {  //  RedundantCondition - Found a redundant condition when evaluating $changed
        echo "Changed\n";
        return true;
    }   

    return false;
}
```

**Comment by muglug** [comment]
> Code like this is rare

That's the best kind
